Utils/src/PhotonIDisoProducer.cc

```diff
@@ -77,28 +77,49 @@ class PhotonIDisoProducer : public edm::global::EDProducer<> {
   edm::EDGetTokenT<double> rhoTok_;
   edm::EDGetTokenT<edm::View<reco::GenParticle>> genParTok_;
   bool debug;
+  effArea effAreas;
+  std::vector<double> effArEtaLow_,effArEtaHigh_; //|eta| boundaries for effective areas
+  std::vector<double> effArChHad_,effArNuHad_,effArGamma_; //effective area values for each of the |eta| ranges
+  double hadTowOverEm_EB_cut_, sieie_EB_cut_, pfChIsoRhoCorr_EB_cut_;
+  double hadTowOverEm_EE_cut_, sieie_EE_cut_, pfChIsoRhoCorr_EE_cut_;
+  std::vector<double> pfNuIsoRhoCorr_EB_cut_, pfNuIsoRhoCorr_EE_cut_; //Rho corrected PF neutral ISO is calulated as [0] + [1]*pho_pt + [2]*pho_pt^2
+  std::vector<double> pfGmIsoRhoCorr_EB_cut_, pfGmIsoRhoCorr_EE_cut_; //Rho corrected PF gamma ISO is calulated as [0] + [1]*pho_pt
 };
 
 
-PhotonIDisoProducer::PhotonIDisoProducer(const edm::ParameterSet& iConfig):
-  photonCollection(iConfig.getUntrackedParameter<edm::InputTag>("photonCollection")),
-  electronCollection(iConfig.getUntrackedParameter<edm::InputTag>("electronCollection")),
-  conversionCollection(iConfig.getUntrackedParameter<edm::InputTag>("conversionCollection")),
-  beamspotCollection(iConfig.getUntrackedParameter<edm::InputTag>("beamspotCollection")),
-  ecalRecHitsInputTag_EE_(iConfig.getParameter<edm::InputTag>("ecalRecHitsInputTag_EE")),
-  ecalRecHitsInputTag_EB_(iConfig.getParameter<edm::InputTag>("ecalRecHitsInputTag_EB")),
-  rhoCollection(iConfig.getUntrackedParameter<edm::InputTag>("rhoCollection")),
-  genParCollection(iConfig.getUntrackedParameter<edm::InputTag>("genParCollection")),
-  photonTok_(consumes<edm::View<pat::Photon>>(photonCollection)),
-  electronTok_(consumes<pat::ElectronCollection>(electronCollection)),
-  conversionTok_(consumes<std::vector<reco::Conversion>>(conversionCollection)),
-  beamspotTok_(consumes<reco::BeamSpot>(beamspotCollection)),
-  ecalRecHitsInputTag_EE_Token_(consumes<EcalRecHitCollection>(ecalRecHitsInputTag_EE_)),
-  ecalRecHitsInputTag_EB_Token_(consumes<EcalRecHitCollection>(ecalRecHitsInputTag_EB_)),
-  rhoTok_(consumes<double>(rhoCollection)),
-  genParTok_(consumes<edm::View<reco::GenParticle>>(genParCollection)),
-  debug(iConfig.getUntrackedParameter<bool>("debug",true))
-{
+PhotonIDisoProducer::PhotonIDisoProducer(const edm::ParameterSet& iConfig){
+  photonCollection       = iConfig.getUntrackedParameter<edm::InputTag>("photonCollection");
```

Comment:
this is the exact opposite of what I wanted you to do

Reply:
I looked at what Alex was pointing to and made changes according to that : https://github.com/aperloff/TreeMaker/blob/2fc19aff65d33facd109443b5fd10d9bb3f67220/Utils/src/GoodJetsProducer.cc#L85-L118

Reply:
well, it should be improved there too. Please undo this change and put all your new parameters into the initialization list (this comes *before* the `{` in the constructor).

Reply:
I could implement for vector<double>. But for double it gives the error when I tried this:
`debug(iConfig.getUntrackedParameter<bool>("debug",true)),
  hadTowOverEm_EB_cut_(iConfig.getParameter<double>("hadTowOverEm_EB_cut")),
`
While compiling it gives this error: "/uscms_data/d3/vhegde/Physics/makeRA2Trees_V13/for2017/CMSSW_8_0_30/src/TreeMaker/Utils/src/PhotonIDisoProducer.cc: In constructor 'PhotonIDisoProducer::PhotonIDisoProducer(const edm::ParameterSet&)':
/uscms_data/d3/vhegde/Physics/makeRA2Trees_V13/for2017/CMSSW_8_0_30/src/TreeMaker/Utils/src/PhotonIDisoProducer.cc:84:10: error: 'PhotonIDisoProducer::hadTowOverEm_EE_cut_' will be initialized after [-Werror=reorder]
   double hadTowOverEm_EE_cut_, sieie_EE_cut_, pfChIsoRhoCorr_EE_cut_;"
........
Any idea why it is so?

Reply:
Exactly what the error says: you have to initialize in the order the member variables are declared in the class.

Reply:
I never expected that the order matters.
This is how the final code looks like: 
`pfGmIsoRhoCorr_EB_cut_(iConfig.getParameter <std::vector<double>> ("pfGmIsoRhoCorr_EB_cut")),
 pfGmIsoRhoCorr_EE_cut_(iConfig.getParameter <std::vector<double>> ("pfGmIsoRhoCorr_EE_cut"))
{`
Is there any other changes that are needed? Should I just commit these changes? If so, any specific message for the commit?

Reply:
If you have made the change to put everything that was in the initializer list back in the initializer list, and to add all of the new parameters (not just those two you listed), please commit it.